commands/src/main/java/org/wildfly/extras/creaper/commands/security/realms/AddTruststoreAuthentication.java

```diff
@@ -0,0 +1,116 @@
+                .parameter("replaceExisting", replaceExisting)
+                .build());
+    }
+
+    @Override
+    public final void apply(OnlineCommandContext ctx) throws Exception {
+        Address truststoreAuthAddress = securityRealmAddress.and("authentication", "truststore");
+
+        Operations ops = new Operations(ctx.client);
+
+        if (replaceExisting) {
+            boolean truststoreAuthExist = ops.exists(truststoreAuthAddress);
+            if (truststoreAuthExist) {
+                ops.remove(truststoreAuthAddress);
+            }
+            new Administration(ctx.client).reloadIfRequired();
+        }
+
+        ops.add(truststoreAuthAddress, Values.empty()
+                .and("keystore-password", truststorePassword)
+                .andOptional("keystore-path", truststorePath)
+                .andOptional("keystore-provider", truststoreProvider)
+                .andOptional("keystore-relative-to", truststoreRelativeTo));
+    }
+
+
+    public static final class Builder extends AbstractAddSecurityRealmSubElement.Builder<Builder> {
+
+        private String truststorePassword;
+        private String truststorePath;
+        private String truststoreProvider;
+        private String truststoreRelativeTo;
+
+        public Builder(String securityRealmName) {
+            super(securityRealmName);
+        }
+
+        /**
+         * Defines the password to open the truststore. It is mandatory parameter when also defining truststore
+         */
+        public Builder truststorePassword(String truststorePassword) {
+            this.truststorePassword = truststorePassword;
+            return this;
+        }
+
+        /**
+         * Defines the path of the trustore, will be ignored if the truststore provider is anything other than JKS.
+         * If not defined, truststore is not defined for the server.
+         */
+        public Builder truststorePath(String truststorePath) {
+            this.truststorePath = truststorePath;
+            return this;
+        }
+
+        /**
+         * Defines the provider for loading the truststore, defaults to JKS.
+         */
+        public Builder truststoreProvider(String truststoreProvider) {
+            this.truststoreProvider = truststoreProvider;
+            return this;
+        }
+
+        /**
+         * Define name of another previously named path, or of one of the standard paths provided by the system.
+         * If 'relative-to' is provided, the value of the 'path' attribute is treated as relative to the path
+         * specified by this attribute.
+         */
+        public Builder truststoreRelativeTo(String truststoreRelativeTo) {
+            this.truststoreRelativeTo = truststoreRelativeTo;
+            return this;
+        }
+
+        @Override
+        public AddTruststoreAuthentication build() {
+            if (truststorePath != null && truststorePassword == null) {
```

Comment:
truststorePath is optional, for example in case of PKCS11 you dont have to specify truststorePath. Please leave just  `if (truststorePassword == null) {` with message "truststorePassword is manadatory when defining truststore"